contracts/IRelayRecipient.sol

```diff
@@ -32,7 +32,20 @@ contract IRelayRecipient {
      *  @param signature - sender's signature over all parameters except approvalData
      *  @param approvalData - extra dapp-specific data (e.g. signature from trusted party)
      */
-    function acceptRelayedCall(address relay, address from, bytes memory encodedFunction, uint gasPrice, uint transactionFee, bytes memory signature, bytes memory approvalData) public view returns (uint);
+     function acceptRelayedCall(
+        address relay,
+        address from,
+        bytes memory encodedFunction,
+        uint256 transactionFee,
+        uint256 gasPrice,
+        uint256 gasLimit,
+        uint256 nonce,
+        bytes memory approvalData,
+        uint256 maxPossibleCharge
```

Comment:
To remove the need to recalculate it in the client. Since we are now only passing these parameters in `accept`, and it is a relatively small value, the gas cost of pushing it to the client is very small (comparable to the cost of actually recalculating it). I'd agree on removing it if it meant the difference between using the EncoderV2 or not, but since it doesn't change much, I'd keep it.